airsonic-main/src/main/java/org/airsonic/player/service/search/SearchServiceUtilities.java

```diff
@@ -1,200 +1,187 @@
+    @Autowired
+    private ArtistDao artistDao;
+
+    /* Search by id only. */
+    @Autowired
+    private AlbumDao albumDao;
+
+    /*
+     * Search by id only.
+     * Although there is no influence at present,
+     * mediaFileService has a caching mechanism.
+     * Service is used instead of Dao until you are sure you need to use mediaFileDao.
+     */
+    @Autowired
+    private MediaFileService mediaFileService;
+
+    public final Function<Long, Integer> round = (i) -> {
+        // return
+        // NumericUtils.floatToSortableInt(i);
+        return i.intValue();
+    };
+
+    public final Function<Document, Integer> getId = d -> {
+        return Integer.valueOf(d.get(FieldNames.ID));
+    };
+
+    public final BiConsumer<List<MediaFile>, Integer> addMediaFileIfAnyMatch = (dist, id) -> {
+        if (!dist.stream().anyMatch(m -> id == m.getId())) {
+            MediaFile mediaFile = mediaFileService.getMediaFile(id);
+            if (!isEmpty(mediaFile)) {
+                dist.add(mediaFile);
+            }
+        }
+    };
+
+    public final BiConsumer<List<Artist>, Integer> addArtistId3IfAnyMatch = (dist, id) -> {
+        if (!dist.stream().anyMatch(a -> id == a.getId())) {
+            Artist artist = artistDao.getArtist(id);
+            if (!isEmpty(artist)) {
+                dist.add(artist);
+            }
+        }
+    };
+
+    public final Function<Class<?>, @Nullable IndexType> getIndexType = (assignableClass) -> {
+        IndexType indexType = null;
+        if (assignableClass.isAssignableFrom(Album.class)) {
+            indexType = IndexType.ALBUM_ID3;
+        } else if (assignableClass.isAssignableFrom(Artist.class)) {
+            indexType = IndexType.ARTIST_ID3;
+        } else if (assignableClass.isAssignableFrom(MediaFile.class)) {
+            indexType = IndexType.SONG;
+        }
+        return indexType;
+    };
+
+    public final Function<Class<?>, @Nullable String> getFieldName = (assignableClass) -> {
+        String fieldName = null;
+        if (assignableClass.isAssignableFrom(Album.class)) {
+            fieldName = FieldNames.ALBUM;
+        } else if (assignableClass.isAssignableFrom(Artist.class)) {
+            fieldName = FieldNames.ARTIST;
+        } else if (assignableClass.isAssignableFrom(MediaFile.class)) {
+            fieldName = FieldNames.TITLE;
+        }
+        return fieldName;
+    };
+
+    public final BiConsumer<List<Album>, Integer> addAlbumId3IfAnyMatch = (dist, subjectId) -> {
+        if (!dist.stream().anyMatch(a -> subjectId == a.getId())) {
+            Album album = albumDao.getAlbum(subjectId);
+            if (!isEmpty(album)) {
+                dist.add(album);
+            }
+        }
+    };
+
+    private final Function<String, File> getRootDirectory = (version) -> {
+        return new File(SettingsService.getAirsonicHome(), version);
+    };
+
+    public final BiFunction<String, IndexType, File> getDirectory = (version, indexType) -> {
```

Comment:
Some of these functions such as this one should probably be package private (remove `public` keyword). Actually I'd make this whole class package private to really drive home the point that it shouldn't be exposed and is an implementation detail.

Reply:
I revised that according to your advice.